File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l, r = 0, len(nums) - 1
        while l <= r:
            m = (l+r) // 2
            print(m)
            if nums[m] > nums[-1]:
                l = m + 1
            elif nums[m] < nums[-1] and nums[m-1] < nums[-1]:
                r = m - 1
            else:
                return nums[m]

File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> complement;
        for (int i = 0; i < nums.size()-1; i++) {
            int remain = target - nums[i];
            if (complement.contains(remain)) {
                return {complement[remain], i};
            }
            complement[nums[i]] = i;
        }
    }
};
